USB basic and hid greentea test cleanup and enable to the bare metal profile

rajkan01 opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html).

NOTE: Do not remove any of the template headings (even for optional sections) as this
template is automatically parsed. 
-->

### Summary of changes <!-- Required -->
- Enabled USB basic and hid test to bare metal profile.
- Removed the partially (The green tea test host side implementation is missing on pyusb_basic.py ) implemented USB control transfer `VENDOR_TEST_CTRL_NONE_DELAY` and `VENDOR_TEST_CTRL_NONE` request type as it is becoming obsolete.
- Removed the USB control request type `VENDOR_TEST_CTRL_IN_DELAY`, `VENDOR_TEST_CTRL_OUT_DELAY`, `VENDOR_TEST_CTRL_IN_STATUS_DELAY`, `VENDOR_TEST_CTRL_OUT_STATUS_DELAY` macros as did not have any test.
- Removed high priority event queue which is used by `VENDOR_TEST_CTRL_NONE_DELAY` request type test. 

<!-- 
    Please provide the following information: 

    Description of the change (what is this fixing / adding / removing?).

    Why the change is needed (if this is fixing a reported issue please summarize what
    the issue is and add the reference. E.g. Fixes #17119).

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
    
-->

#### Impact of changes <!-- Optional -->
With these changes, USB basic and hid greentea tests build and run successfully.
<!-- 
    If there are any implications for users taking this change then they must be 
    provided here. For Major PR types this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

#### Migration actions required <!-- Optional -->
None.
<!-- 
    This should only be applicable in Major PR types for which this field is MANDATORY.

    NOTE: This section is automatically written to release notes for Feature and 
    Major releases and should contain enough details for a user.
-->

### Documentation <!-- Required -->
None.
<!-- 
    Please provide details of any document updates required, including links to any
    related PRs against the docs repository.
    If no document updates are required please specify 'None', this at least tells us
    that this has been considered.
-->

----------------------------------------------------------------------------------------------------------------
### Pull request type <!-- Required -->

<!--
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front of them would change the meaning incorrectly. 
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results <!-- Required -->

<!--
    Provide all the information required, listing all the testing performed. For new targets please attach full test results for all supported compilers.
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers <!-- Optional -->
@evedon @fkjagodzinski @maciejbocianski 
<!--
    Request additional reviewers with @username or @team
-->

----------------------------------------------------------------------------------------------------------------

rajkan01 committed:
USB hid and basic test

- Enable the USB hid and basic test for bare metal profile.
- Remove the partially (The green tea test host side implementation is missing on pyusb_basic.py ) implemented USB control transfer VENDOR_TEST_CTRL_NONE_DELAY request type test case as it is becoming obsolete.
- Remove high priority event queue which is used by that request type test case.

evedon commented on TESTS/usb_device/basic/USBTester.h:
```diff
@@ -27,9 +27,11 @@
 
 #include "USBDevice.h"
 
+using namespace std::chrono_literals;
```
You should move this to the source file USBtester.cpp where it is needed.

kjbracey commented on TESTS/usb_device/basic/USBTester.cpp:
```diff
@@ -174,7 +177,7 @@ void USBTester::callback_request(const setup_packet_t *setup)
     }
 
-    if (delay) {
+    if (delay != std::chrono::duration<int, std::milli>::zero()) {
```
If you want to use the method, better to just say `delay.zero()`. But you may as well just say `0s` in normal life, I think. Using the `zero()` method is only really worth it where it's important to maintain the specific type.

Yes `0s` is 64-bit, so it does make it a `!=` comparison of a 32-bit millisecond count to a 64-bit zero, but the compiler should, I hope, realise it doesn't actually need to do any 64-bit work for that. Maybe you could check if `!= 0s` is bigger than `!= delay.zero()` on each toolchain? Put my mind at rest.

(But as this is test code, I wouldn't even sweat 32-bit/64-bit anyway. I'd have just made delay be `milliseconds` (ie 64-bit) to save typing.)

And you already used `0s` above when you initialised it, so you're not being consistent!

kjbracey commented on TESTS/usb_device/basic/USBTester.cpp:
```diff
@@ -60,7 +59,11 @@ USBTester::USBTester(USBPhy *phy, uint16_t vendor_id, uint16_t product_id, uint1
     MBED_ASSERT(resolver.valid());
+#if defined(MBED_CONF_RTOS_PRESENT)
     queue = mbed::mbed_highprio_event_queue();
+#else
```
Is there any record in the logs of why it was using high priority specifically? I guess it's working as normal.

If there's no downside, I'd make it normal priority always. If there is a downside, put a comment about it here.

kjbracey commented on TESTS/usb_device/basic/USBTester.cpp:
```diff
@@ -45,6 +43,7 @@
 
 #define EVENT_READY (1 << 0)
 
+using namespace std::chrono_literals;
```
Don't forget you can `using namespace std::chrono` instead to save typing `std::chrono::duration`. or `std::chrono::milliseconds`.

0xc0170 commented on TESTS/usb_device/basic/USBEndpointTester.cpp:
```diff
@@ -173,7 +171,6 @@ USBEndpointTester::USBEndpointTester(USBPhy *phy, uint16_t vendor_id, uint16_t p
         MBED_ASSERT(_endpoint_buffs[i] != NULL);
     }
     MBED_ASSERT(resolver.valid());
-    queue = mbed::mbed_highprio_event_queue();
```
by removing the queue here, the test does not change - it has not been used at all?

rajkan01 replied:
yes. This queue initially added by Russ to create a utility function but the scope of this work became obsolete. This queue is not used any of the test cases on full profile.

0xc0170 commented on TESTS/usb_device/basic/USBTester.cpp:
```diff
@@ -32,7 +30,6 @@
 #define VENDOR_TEST_CTRL_NONE               3
 #define VENDOR_TEST_CTRL_IN_DELAY           4
 #define VENDOR_TEST_CTRL_OUT_DELAY          5
-#define VENDOR_TEST_CTRL_NONE_DELAY         6
```
why is this delay removed? Shall this be a series of commits rather than "enable usb basic" - does not provide a reason for some changes I am commenting now on

0xc0170 replied:
> Removed the USB endpoint test class unused variable and partially (The greentea test host side implementation is missed on pyusb_basic.py ) implemented USB request control transfer of VENDOR_TEST_CTRL_NONE_DELAY so it can run with the bare metal profile.

This would explain some changes, should be in the commit msg as well.

rajkan01 replied:
@0xc0170 Done. Please check

evedon replied:
I don't think it is worth deleting VENDOR_TEST_CTRL_NONE_DELAY (here and in the .py file) because it breaks the defines and it is not the only one not implemented. For example I don't see any code calling VENDOR_TEST_CTRL_IN_STATUS_DELAY or VENDOR_TEST_CTRL_OUT_STATUS_DELAY

rajkan01 replied:
After discussion with @evedon have decided to remove all unused USB control transfer request type and also partially implemented test.

rajkan01 committed:
Cleanup
- Remove the partially implemented VENDOR_TEST_CTRL_NONE USB control request type.
- Remove the USB control request type VENDOR_TEST_CTRL_IN_DELAY, VENDOR_TEST_CTRL_OUT_DELAY, VENDOR_TEST_CTRL_IN_STATUS_DELAY, VENDOR_TEST_CTRL_OUT_STATUS_DELAY macros as did not have any test.

evedon commented on TESTS/usb_device/basic/USBEndpointTester.cpp:
```diff
@@ -39,10 +37,10 @@
+#define VENDOR_TEST_CTRL_IN_SIZES       3
+#define VENDOR_TEST_CTRL_OUT_SIZES      4
+#define VENDOR_TEST_RW_RESTART          5
+#define VENDOR_TEST_ABORT_BUFF_CHECK    6
```
We could put these declarations in a header so there is no duplication between this source file and USBTester.cpp but let's go with this PR as is now.
